### ISSUE TITLE
Stories for cookieplone usage

### ISSUE THREAD

**Issue by stevepiercy** [plone/cookieplone#22]
Stories for using cookieplone. Please edit this description as needed.

- [ ] Develop Plone backend and contribute to its core
- [ ] Develop Plone frontend Volto and contribute to its core
- [ ] Create a project with a backend only
- [ ] Create a project with a frontend only
- [ ] Create a project with both a backend and frontend
- [ ] Develop a Plone add-on
- [ ] Develop a Volto add-on

Stories for NOT using cookieplone.

- [x] I want to demo Plone. Instead visit one of Plone's demos, or use Docker Compose to run your own instance locally.

@jensens @ericof @sneridagh @mauritsvanrees

**Comment by jensens** [comment]
I am not sure if " Develop Plone backend and contribute to its core" is a story for cookieplone. It's still: use [buildout.]coredev

**Comment by davisagli** [comment]
> I am not sure if " Develop Plone backend and contribute to its core" is a story for cookieplone. It's still: use [buildout.]coredev

I agree. For this, use buildout.coredev (if you need easy access to all backend packages) OR the makefile commands in the specific package that needs changes (if it's just one and you know what it is). The cookieplone templates don't currently provide an easy way to check out a specific backend package from the correct repository and branch for a given release series of Plone.

> Stories for NOT using cookieplone. I want to demo Plone. Instead visit one of Plone's demos, or use Docker Compose to run your own instance locally.

The recommendation to visit one of the demos makes sense for most people in this scenario. I would not recommend using Docker Compose for this. It's a separate set of install documentation that needs to be written, maintained, and clarified when to use it. And it's a dead end once you've completed your demo and want to actually work on changing something.

One thing I want, which I'm not sure if it's included in the use cases already listed, is a way to create a new project using cookieplone that is based on a specific existing Plone distribution.

**Comment by stevepiercy** [comment]
@jensens @davisagli I made the following edits:

1. Rephrased the purpose of the list for using cookieplone to "Potential stories for using cookieplone, whether or not they currently exist. This is a wish list."
2. Moved "Develop Plone backend and contribute to its core" to the "not use" list, and appended to use `buildout.coredev` instead.
3. Split the demo story into two stories and clarified the situation.

> One thing I want, which I'm not sure if it's included in the use cases already listed, is a way to create a new project using cookieplone that is based on a specific existing Plone distribution.

Can you help me clarify the existing stories to be more explicit? I'm not clear how this is story is distinct from the current stories?

Thanks for your feedback!

**Comment by davisagli** [comment]
> I'm not clear how this is story is distinct from the current stories?

@stevepiercy I'm not sure if it's distinct or not! I guess the default projects come with both the Volto and Classic distributions available, and I can pick which one I want in the UI when I create the Plone site. But what's the intended path if I want to create a project based on another distribution like the intranet or EDU distributions? /cc @ericof

**Comment by animus888** [comment]
> Potential stories for using cookieplone, whether or not they currently exist. This is a wish list. Please edit this description as needed.
> 
>    Develop Plone frontend Volto and contribute to its core
> 
>    Create a project with a backend only
> 
>    Create a project with a frontend only
> 
>    Create a project with both a backend and frontend
> 
>    Develop a Plone add-on
> 
>    Develop a Volto add-on
> 
What about ....
* Create a project with a frontend  and a distribution (backend) ?

**Comment by stevepiercy** [comment]
@animus888 @davisagli it sounds like you're both thinking similar things. Looking at the definition of [Plone distribution](https://github.com/plone/plone.distribution?tab=readme-ov-file#what-is-a-plone-distribution):

> A Plone distribution is a pre-packaged version of Plone that includes specific features, themes, modules, and configurations. It is a convenient way to get a specific type of website up and running quickly, as the distribution includes everything needed to run that type of site.

That definition does not specify whether it includes a frontend or backend, but it sounds like both are always included.

With that, I would suggest we add:

- Create a project based upon a [Plone distribution](https://github.com/plone/plone.distribution?tab=readme-ov-file#what-is-a-plone-distribution).

I would also suggest that we define which frontends and backends are available for each story. That could be a preface to the list, such as "Available frontends: Classic UI, Volto, [other]".
